src/wui/minimap.cc

```diff
@@ -98,8 +98,10 @@ void MiniMap::View::set_zoom(const bool zoom) {
 bool MiniMap::View::can_zoom() {
 	const Widelands::Map& map = ibase_.egbase().map();
 	// The zoomed MiniMap needs to fit into: height - windows boarders - button height. -> 60px
-	return !(scale_map(map, true) == 1 ||
-	         map.get_height() * scale_map(map, true) > ibase_.get_h() - 60);
+	auto scale = scale_map(map, true);
+	return !(scale == 1 ||
+		 map.get_width() * scale > ibase_.get_w() - 60 ||
+		 map.get_height() * scale > ibase_.get_h() - 60);
```

Comment:
Maybe it's just me, but I think this would be easier to understand without the negation (`scale != 1 && foo <= bar && foo <= bar`)

Reply:
Yes, it is easier and also no need to check  `foo <= bar`  if `scale != 1` fails. Going even further we should use `scale > 1` but not really necessary because `scale_map` function always returns  >= 1.